src/module.c

```diff
@@ -8805,6 +8805,71 @@ int RM_ACLAddLogEntry(RedisModuleCtx *ctx, RedisModuleUser *user, RedisModuleStr
     return REDISMODULE_OK;
 }
 
+/* This is to prevent a coupling between redis internals and modules */
+uint64_t mapInternalScriptFlagsToModuleFlags(uint64_t flags) {
+    uint64_t flags_out = 0;
+
+    if (flags & SCRIPT_FLAG_NO_WRITES) {
+        flags_out |= RM_SCRIPT_FLAG_NO_WRITES;
+    }
+
+    if (flags & SCRIPT_FLAG_ALLOW_OOM) {
+        flags_out |= RM_SCRIPT_FLAG_ALLOW_OOM;
+    }
+
+    if (flags & SCRIPT_FLAG_ALLOW_STALE) {
+        flags_out |= RM_SCRIPT_FLAG_ALLOW_STALE;
+    }
+
+    if (flags & SCRIPT_FLAG_NO_CLUSTER) {
+        flags_out |= RM_SCRIPT_FLAG_NO_CLUSTER;
+    }
+
+    if (flags & SCRIPT_FLAG_EVAL_COMPAT_MODE) {
+        flags_out |= RM_SCRIPT_FLAG_EVAL_COMPAT_MODE;
+    }
+
+    if (flags & SCRIPT_FLAG_ALLOW_CROSS_SLOT) {
+        flags_out |= RM_SCRIPT_FLAG_ALLOW_CROSS_SLOT;
+    }
+
+    return flags_out;
+}
+
+RedisModuleString * RM_GetScriptBodyFlags(RedisModuleCtx *ctx, RedisModuleString *body, uint64_t *flags_out) {
+    sds err = NULL;
+    uint64_t flags;
+
+    if (getScriptFlags(NULL, body->ptr, &flags, &err) == C_ERR) {
+        RedisModuleString *ret = RM_CreateString(ctx, err, sdslen(err));
+        sdsfree(err);
+        return ret;
+    }
+
+    if (flags_out != NULL) {
```

Comment:
If `flags_out` is NULL, I think it is a bug/misuse, wdyt?. Why would someone call this function with a NULL ptr? I think we should verify this at the beginning and return an error if it is NULL.